Mathlib/Analysis/NormedSpace/Basic.lean

```diff
@@ -341,22 +342,52 @@ protected theorem NormedSpace.unbounded_univ : ¬Bornology.IsBounded (univ : Set
   hx.not_le (hR x trivial)
 #align normed_space.unbounded_univ NormedSpace.unbounded_univ
 
-/-- A normed vector space over a nontrivially normed field is a noncompact space. This cannot be
-an instance because in order to apply it, Lean would have to search for `NormedSpace 𝕜 E` with
-unknown `𝕜`. We register this as an instance in two cases: `𝕜 = E` and `𝕜 = ℝ`. -/
-protected theorem NormedSpace.noncompactSpace : NoncompactSpace E :=
-  ⟨fun h => NormedSpace.unbounded_univ 𝕜 _ h.isBounded⟩
+protected lemma NormedSpace.cobounded_neBot : NeBot (cobounded E) := by
+  rw [neBot_iff, Ne.def, cobounded_eq_bot_iff, ← isBounded_univ]
+  exact NormedSpace.unbounded_univ 𝕜 E
+
+instance (priority := 100) NontriviallyNormedField.cobounded_neBot : NeBot (cobounded 𝕜) :=
+  NormedSpace.cobounded_neBot 𝕜 𝕜
+
+instance (priority := 100) RealNormedSpace.cobounded_neBot [NormedSpace ℝ E] :
+    NeBot (cobounded E) := NormedSpace.cobounded_neBot ℝ E
+
+instance (priority := 100) NontriviallyNormedField.infinite : Infinite 𝕜 :=
+  ⟨fun _ ↦ NormedSpace.unbounded_univ 𝕜 𝕜 (Set.toFinite _).isBounded⟩
+
+end NontriviallyNormedSpace
+
+variable (𝕜 E : Type*) [NormedField 𝕜] [Infinite 𝕜] [NormedAddCommGroup E] [Nontrivial E]
+  [NormedSpace 𝕜 E]
+
+/-- A normed vector space over an infinite normed field is a noncompact space.
+This cannot be an instance because in order to apply it,
+Lean would have to search for `NormedSpace 𝕜 E` with unknown `𝕜`.
+We register this as an instance in two cases: `𝕜 = E` and `𝕜 = ℝ`. -/
+protected theorem NormedSpace.noncompactSpace : NoncompactSpace E := by
+  by_cases H : ∃ c : 𝕜, c ≠ 0 ∧ ‖c‖ ≠ 1
+  · letI := NontriviallyNormedField.ofNormNeOne H
+    exact ⟨fun h ↦ NormedSpace.unbounded_univ 𝕜 E h.isBounded⟩
+  · push_neg at H
+    rcases exists_ne (0 : E) with ⟨x, hx⟩
+    suffices ClosedEmbedding (Infinite.natEmbedding 𝕜 · • x) from this.noncompactSpace
```

Comment:
That would be good to have these closed embeddings in general? and even in finite dimension.

Reply:
What general statement are you talking about? https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Module/FiniteDimension.html#closedEmbedding_smul_left ? Another statement is that a trivially normed field has discrete topology (so, in fact we can skip `Infinite.natEmbedding 𝕜`).

Reply:
I was thinking of the `suffices` that you prove on the next lines. I guess it also holds when the field is nontrivially valued. 
Actually, I am not sure you need to make the cases disjoint: once you reduce to the case of a line by these closed embeddings, what matters is that `k` is noncompact.

Reply:
If `k = Q`, then this scalar multiplication is not a **closed** embedding.